vega_sim/scenario/fairground/scenario.py

```diff
@@ -0,0 +1,433 @@
+from typing import Optional, Union
+
+from vega_sim.scenario.scenario import Scenario
+from vega_sim.null_service import VegaService
+from vega_sim.null_service import VegaServiceNull
+from vega_sim.environment.environment import (
+    MarketEnvironmentWithState,
+    NetworkEnvironment,
+)
+from vega_sim.environment.agent import Agent
+from vega_sim.scenario.common.utils.price_process import (
+    get_historic_price_series,
+    Granularity,
+)
+from vega_sim.scenario.common.agents import (
+    MarketManager,
+    ExponentialShapedMarketMaker,
+    OpenAuctionPass,
+    MarketOrderTrader,
+    MomentumTrader,
+    PriceSensitiveMarketOrderTrader,
+)
+from vega_sim.scenario.fairground.agents import (
+    WALLET_NAME,
+    WALLET_PASS,
+    MARKET_MANAGER_KEY,
+    MARKET_MAKER_KEY,
+    AUCTION_PASS_KEYS,
+    RANDOM_MARKET_ORDER_AGENT_KEYS,
+    MOMENTUM_MARKET_ORDER_AGENT_KEYS,
+    SENSITIVE_MARKET_ORDER_AGENT_KEYS,
+)
+
+
+class LivePrice:
+    """Iterator for getting a live product price process.
+
+    Class is to be used when running the scenario on fairground incentives. The
+    iterator can be passed to the market-maker agent and the price-sensitive
+    agents to give them information regarding the live product price.
+
+    """
+
+    def __init__(self, product: str = "ADAUSDT"):
+        self.product = product
+
+    def __iter__(self):
+        return self
+
+    def __getitem__(self, index):
+        return self._get_price()
+
+    def __next__(self):
+        return self._get_price()
+
+    def _get_price(self):
+        url = f"https://api.binance.com/api/v3/avgPrice?symbol={self.product}"
+        return float(requests.get(url).json()["price"])
+
+
+# Set default scenario arguments
+PRICE_PROCESS_ARGS = {
+    "binance_product": "ADAUSDT",
+    "coinbase_product": "ADA-USD",
+    "start": "2022-08-01 00:00:00",
+    "randomise": False,
+}
+MARKET_MANAGER_ARGS = {
+    "initial_mint": 1e08,
+    "commitment_amount": 1e02,
+    "market_name": "ADA Monthly (Historic)",
+    "asset_name": "tUSD",
+    "adp": 4,
+    "mdp": 4,
+    "pdp": 4,
+}
+MARKET_MAKER_ARGS = {
+    "initial_asset_mint": 1e08,
+    "commitment_amount": 1e03,
+    "kappa": 1000,
+    "market_kappa": 2000,
+    "market_order_arrival_rate": 400,
+    "num_levels": 20,
+    "tick_spacing": 0.001,
+    "order_unit_size": 100,
+    "inventory_upper_boundary": 80,
+    "inventory_lower_boundary": -80,
+}
+AUCTION_PASS_ARGS = {
+    "initial_asset_mint": 1e06,
+    "side": ["SIDE_BUY", "SIDE_SELL"],
+    "initial_price": None,
+    "opening_auction_trade_amount": 10,
+}
+RANDOM_MARKET_ORDER_ARG = {
+    "initial_asset_mint": 1e06,
+    "buy_intensity": 1,
+    "sell_intensity": 1,
+}
+MOMENTUM_MARKET_ORDER_ARGS = {
+    "initial_asset_mint": 1e06,
+    "momentum_strategies": ["MACD", "APO", "RSI", "STOCHRSI", "CPO"],
+    "order_intensity": 5,
+}
+SENSITIVE_MARKET_ORDER_ARGS = {
+    "initial_asset_mint": 1e06,
+    "buy_intensity": 5,
+    "sell_intensity": 5,
+}
```

Comment:
It would be nice to move these out to a JSON config file so someone can tweak and pass in without having to edit the code itself, but not in this PR :)

Reply:
Yeah I agree, I think it would be really nice if we didn't have to pass a huge number of arguments to a scenario but could instead pass in a single argument for which config to use for each agent type.

A bit like in the vega feature tests where you have configs you can use to create a market quickly.